mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -456,25 +456,31 @@ fn init_wasm_event_streaming(ctx: &MmArc) {
 
 #[cfg(not(target_arch = "wasm32"))]
 async fn init_db_migration_watcher_loop(ctx: MmArc) {
+    use std::collections::HashSet;
+
+    let mut migrations = HashSet::new();
     let db_migration_watcher = &ctx
         .init_db_migration_watcher()
         .await
-        .expect("db_migration_watcher initialization failed");
+        .expect("db_m igration_watcher initialization failed");
 
-    let watcher_clone = db_migration_watcher.clone();
     let receiver = db_migration_watcher.get_receiver();
     let mut guard = receiver.lock().await;
```

Comment:
So I also think we can return the receiver here and not store it in the watcher/sender all together. so the `init` method will return `(Sender, Receiver)`. `Sender` gets stored in the `ctx` (as the db_migration_watcher) and the receiver stored as a local var in the watching loop.

Reply:
thanks done!